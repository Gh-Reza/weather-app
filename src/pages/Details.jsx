import { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { useSelector } from 'react-redux/es/hooks/useSelector';
import { SpringSpinner } from 'react-epic-spinners';
import { getWeather, resetStore } from '../redux/weather/weatherSlice';
import State from '../components/State';

const Details = () => {
  const states = useSelector((store) => store.countryStates.countryStates);
  const details = useSelector((store) => store.weather.weatherDetails);
  const isLoading = useSelector((store) => store.weather.isLoading);
  const res = details.filter((detail) => detail.province !== '');
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(resetStore([]));
    states.forEach((state) => {
      dispatch(getWeather(state.name));
    });
  }, [dispatch, states]);

  if (states.hasError) {
    return (
      <div className="gen container d-flex justify-content-center align-items-center">
        <h2 className="text-danger">Something went wrong</h2>
      </div>
    );
  }

  return (
    <div className="gen container flex-grow-1 d-flex flex-column justify-content-center">
      {isLoading ? (
        <div className="container d-flex justify-content-center align-items-center">
          <SpringSpinner size={60} color="#333" />
        </div>
      ) : (
        res.map((state) => (
          <State
            key={state.id}
            province={state.province}
            icon={state.icon}
            condition={state.condition}
            temp={state.temp}
          />
        ))
      )}
    </div>
  );
};

export default Details;
