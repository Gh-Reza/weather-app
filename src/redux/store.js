import { configureStore, applyMiddleware } from '@reduxjs/toolkit';
import countriesReducer from './countries/countriesSlice';

const store = configureStore({
  reducer: {
    countries: countriesReducer,
  },
}, applyMiddleware);

export default store;
