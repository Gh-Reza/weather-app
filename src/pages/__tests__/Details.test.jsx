import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../redux/store';
import '@testing-library/jest-dom/extend-expect';
import Details from '../Details';

describe('Tests the details page', () => {
  const component = (
    <Provider store={store}>
      <Details />
    </Provider>
  );

  it('Match the snapshot', () => {
    const tree = render(component);
    expect(tree).toMatchSnapshot();
  });
});
