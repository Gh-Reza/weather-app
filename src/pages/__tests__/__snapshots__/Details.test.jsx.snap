// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the details page Match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="gen container mt-4 flex-grow-1 d-flex flex-column justify-content-center"
      >
        <div
          class="container d-flex justify-content-center align-items-center"
        >
          <div
            animationduration="3000"
            class="sc-hAZoWV iCvdcq spring-spinner"
            color="#fff"
            size="60"
          >
            <div
              class="spring-spinner-part top"
            >
              <div
                class="spring-spinner-rotator"
              />
            </div>
            <div
              class="spring-spinner-part bottom"
            >
              <div
                class="spring-spinner-rotator"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="gen container mt-4 flex-grow-1 d-flex flex-column justify-content-center"
    >
      <div
        class="container d-flex justify-content-center align-items-center"
      >
        <div
          animationduration="3000"
          class="sc-hAZoWV iCvdcq spring-spinner"
          color="#fff"
          size="60"
        >
          <div
            class="spring-spinner-part top"
          >
            <div
              class="spring-spinner-rotator"
            />
          </div>
          <div
            class="spring-spinner-part bottom"
          >
            <div
              class="spring-spinner-rotator"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
